tracing-tower/examples/h2-server.rs

```diff
@@ -0,0 +1,156 @@
+use bytes::{Bytes, IntoBuf};
+use futures::*;
+use http::Request;
+use tokio::net::TcpListener;
+use tokio::runtime::Runtime;
+use tower_h2::{Body, RecvBody, Server};
+use tower_service::Service;
+use tracing::Level;
+use tracing_futures::Instrument;
+use tracing_tower::InstrumentMake;
+
+type Response = http::Response<RspBody>;
+
+struct RspBody(Option<Bytes>);
+
+impl RspBody {
+    fn new(body: Bytes) -> Self {
+        RspBody(Some(body))
+    }
+
+    fn empty() -> Self {
+        RspBody(None)
+    }
+}
+
+impl Body for RspBody {
+    type Data = <Bytes as IntoBuf>::Buf;
+    type Error = h2::Error;
+
+    fn poll_data(&mut self) -> Poll<Option<Self::Data>, Self::Error> {
+        let data = self.0.take().and_then(|b| {
+            if b.is_empty() {
+                None
+            } else {
+                Some(b.into_buf())
+            }
+        });
+        Ok(Async::Ready(data))
+    }
+
+    fn poll_trailers(&mut self) -> Poll<Option<http::HeaderMap>, Self::Error> {
+        Ok(None.into())
+    }
+}
+
+const ROOT: &'static str = "/";
+
+#[derive(Debug)]
+struct Svc;
+impl Service<Request<RecvBody>> for Svc {
+    type Response = Response;
+    type Error = h2::Error;
+    type Future = future::FutureResult<Response, Self::Error>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(Async::Ready(()))
+    }
+
+    fn call(&mut self, req: Request<RecvBody>) -> Self::Future {
+        tracing::debug!("received request");
+        let mut rsp = http::Response::builder();
+        rsp.version(http::Version::HTTP_2);
+
+        let uri = req.uri();
+        let rsp = if uri.path() != ROOT {
+            let body = RspBody::empty();
+            tracing::warn!(rsp.error = %"unrecognized path", req.path = ?uri.path());
+            rsp.status(404).body(body).unwrap()
+        } else {
+            let body = RspBody::new("heyo!".into());
+            rsp.status(200).body(body).unwrap()
+        };
+        tracing::debug!(rsp.status = %rsp.status(), message = "sending response");
+        future::ok(rsp)
+    }
+}
+
+#[derive(Debug)]
+struct NewSvc;
+impl tower_service::Service<()> for NewSvc {
+    type Response = Svc;
+    type Error = ::std::io::Error;
+    type Future = future::FutureResult<Svc, Self::Error>;
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(Async::Ready(()))
+    }
+
+    fn call(&mut self, _target: ()) -> Self::Future {
+        future::ok(Svc)
+    }
+}
+
+fn main() {
+    let subscriber = tracing_fmt::FmtSubscriber::builder()
+        .with_filter(tracing_fmt::filter::EnvFilter::from(
+            "h2_server=trace,tracing_tower=trace",
+        ))
+        .finish();
+
+    let _ = tracing::subscriber::set_global_default(subscriber);
+    let mut rt = Runtime::new().unwrap();
+    let reactor = rt.executor();
```

Comment:
This code was originally copied directly from `tower-h2`'s examples dir at revision tower-rs/tower-h2@9b96d8d5eabe56a44a7d01228b14e96d875e84b3, which [used this naming](https://github.com/tower-rs/tower-h2/blob/9b96d8d5eabe56a44a7d01228b14e96d875e84b3/examples/server.rs#L100). The example has since been updated and [is now much simpler](https://github.com/tower-rs/tower-h2/blob/0865040d699697bbaf1c3b77b3f256b72f98cdf4/examples/server.rs), we should probably update this example as well...

Reply:
how about ill update this to hyper :)